tools/eliminate-duplicate-functions.js

```diff
@@ -0,0 +1,489 @@
+///////////////////////////////////////////////////////////////////////////////////////////////
+// Eliminate-Duplicate-Functions.js
+//
+// This is a Javascript file that is used to post-process an Emscripten transpiled JS file.
+// It will remove all the duplicate functions from the generated ASM. In its current form,
+// the input JS file is expected to be an Emscripten generated file of the form:
+//
+// <PRE-ASM-JS-SHELL>
```

Comment:
this is the input format of the `.py` file, which is split up and sent to here in chunks which don't look this this - I think?